### PR TITLE
feat: contract salt

### PR DESCRIPTION
**Pull request by drspacemn**
Allow Contract Salt to be included in Deploy transactions instead of hard coded to default.

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

contract address salt is hardcoded 

Issue Number: #92 

# What is the new behavior?

- contract address salt can be set by the Deploy transaction

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by EvolveArt — APPROVED** [review]
lgtm

**Comment by abdelhamidbakhta** [comment]
Ready to merge when conflict is resolved @drspacemn

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm

File: crates/pallets/starknet/src/mock.rs
Comment (drspacemn) [inline crates/pallets/starknet/src/mock.rs:82]: @EvolveArt this change isn't very descriptive. I'd like to change it before you merge. 

This hash doesn't actually apply to the contract located @ `ACCOUNT_CONTRACT_PATH` so this value essentially just serves as a lookup key, but incrementing it by one is confusing. 

I'll make this more transparant.

Comment (EvolveArt) [reply crates/pallets/starknet/src/mock.rs:82]: Right I think we do not need to have such complicated values but rather simple ones like the [blockifier](https://github.com/tdelabro/blockifier/blob/no_std-support/crates/blockifier/src/test_utils.rs#L34) does.

Comment (drspacemn) [reply crates/pallets/starknet/src/mock.rs]: Okay attempted to simplify the testing and use actual contract class hashes where applicable.

One thing I don't understand is why blockifier testing isn't using the actual contract_class_hash if they can.

Comment (EvolveArt) [reply crates/pallets/starknet/src/mock.rs]: Fair question to be honest not sure I know exactly why

**Review by abdelhamidbakhta — APPROVED** [review]
Lgtm

**Pull request merged**
